### ISSUE TITLE
Thought on adding support for web-worker.

### ISSUE THREAD

**Issue by rayk** [rmawatson/flutter_isolate#30]
A somewhat related background issue: https://github.com/flutter/flutter/issues/33577

It strikes me that extending this plugin to support web-workers would be tidy solution for flutter application to offload concurrent work in a web setting.

**API**
```dart
spawn<T>(void entryPoint(T message), T message) → Future
```
Would kick off ```window.Worker()```

```dart
controlPort ↔ SendPort
```
 would return a port that proxies around  ```MessageChannel()```

```dart
kill() → void
```
would ```close()``` the channel and ```terminate()``` the worker.

Unfortunately  ```pause() → void``` and ```resume() → void```  does not really map nicely to ```MessagePort()```.

*NOTE: The may not work for service workers which can continue to execute after initiating code terminates.*

Would appreciate any feedback about the above idea.. I happy to attempt an implementation if it's appears feasible.

**Comment by deakjahn** [comment]
@rayk No, it isn't feasible right know. The actual problem is with the very first line you mention, `spawn()`. The problem is that the new worker would immediately need to call back to Dart code, using the `entryPoint`. This is the code of the isolate/worker itself. You write it in Dart, it gets compiled and passed to the isolate on mobile but on the web, even if it gets compiled to the very same JavaScript the web worker would need to use, it's not a separate JS that could be passed to the worker. And the worker can't just call back into our main JS code, that's strictly impossible in the browser.

Basically, while on the mobile the main app and the isolates are kept separate by the framework, both are compiled to the same codebase and it's therefore possible for the isolate to call the entry point. On the web, I can't see anything that would allow a package to provide this. The toolchain itself would need to allow this, basically by compiling the code of the isolate into a separate, standalone JS that can be imported by the worker. The Flutter devs have to provide this, there is no other way (unless, of course, you're ready to duplicate whatever you put into the isolat in JavaScript yourself, which *is* possible but goes against the actual need or re-using what's already there).

**Comment by rayk** [comment]
Thanks for the reply and the effort... Just looking at this for the last two minutes, it could helpful.. Certainly in simplify the message handling... I have perviously implemented all of this in JS and from memory the marshalling of messages back and forth was a pain. I shall have a play with _flutter_isloate_web_ in the the next days and give you some feedback.

**Comment by deakjahn** [comment]
I'm currently using it in an app. The isolate part is older code, it works flawlessly. I also added an async implementation that implements the same interface, so it can be used as a drop-in replacement but it doesn't offer any real parallelism. This one works just fine in Flutter Web, too. This means that my current app is genuinely cross platform with the same isolate/worker code for all platforms, only that the web side doesn't yet enjoy the real benefits. Still, it works.
